tools/build/wheel/split_wheel.py

```diff
@@ -0,0 +1,103 @@
+"""Script to build split pytorch wheels
+
+What is split build / why is it important?
+    > Split build is splitting the PyTorch build into a libtorch &
+    > PyTorch python frontend package. This allows us to to publish
+    > both as separate packages and opens up our ability to have users
+    > install different libtorch backends per their PyTorch frontend
+    >
+    > Example: opening up the door to things like:
+    >     pip install torch[cuda]
+    >     pip install torch[rocm]
+    >     pip install torch[cpu]
+    >     etc.
+
+Why does this exist?
+    > Currently our split build requires you to invoke setup.py twice
+    > Which ends up complicating the build process and adds some level
+    > of complexity to our setup.py / build invocation for split builds.
+    > Ideally this script will eventually not be needed but for
+    > development purposes we should have an easy way to invoke this script
+"""
+
+import argparse
+import os
+import sys
+import subprocess
+
+import logging
+from pathlib import Path
+from typing import List, Dict, Optional
+
+logging.basicConfig(
+    level=logging.INFO, format="%(asctime)s - %(levelname)s - %(message)s"
+)
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.DEBUG)
+
+
+# NOTE: This will need to be updated if this script is ever moved
+ROOT_PATH = Path(__file__).parent.parent.parent.parent
+SETUP_PY_PATH = ROOT_PATH / "setup.py"
+
+
+def requirements_installed() -> bool:
+    try:
+        import setuptools  # type: ignore[import-untyped] # noqa: F401
+
+        return True
+    except ImportError:
+        logger.error(
+            "Requirements not installed, run the following command to install:"
+        )
+        logger.error(
+            "    > %s install -r %s/requirements.txt", sys.executable, ROOT_PATH
+        )
+        return False
+
+
+def setup_py(cmd_args: List[str], extra_env: Optional[Dict[str, str]] = None) -> None:
+    if extra_env is None:
+        extra_env = {}
+    cmd = [sys.executable, str(SETUP_PY_PATH), *cmd_args]
+    logger.debug("+ %s", " ".join(cmd))
+    subprocess.run(
+        cmd,
+        # Give the parent environment to the subprocess
+        env={**os.environ, **extra_env},
+        check=True,
+    )
+
+
+def split_build(cmd: str) -> None:
+    logger.info("Running %s for libtorch wheel", cmd)
+    setup_py(
+        [cmd],
+        extra_env={"BUILD_LIBTORCH_WHL": "1", "BUILD_PYTHON_ONLY": "0"},
```

Comment:
Does it need to inherit any other environment variables?  Like the bash script and build ymls set some env vars, are any of those needed?

Reply:
Those should be passed above in the setup_py command through `**os.environ`